contrib/pkg/adm/managedns/enable.go

```diff
@@ -0,0 +1,211 @@
+		Use:   `enable domain1.example.com domain2.example.com ...`,
+		Short: "Enable managed DNS globally for the Hive cluster.",
+		Long:  longDesc,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			log.SetLevel(log.InfoLevel)
+			if err := opt.Complete(cmd, args); err != nil {
+				return
+			}
+			if err := opt.Validate(cmd); err != nil {
+				return
+			}
+			dynClient, err := contributils.GetClient()
+			if err != nil {
+				log.WithError(err).Fatal("error creating kube clients")
+			}
+
+			err = opt.Run(dynClient, args)
+			if err != nil {
+				log.WithError(err).Error("Error")
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opt.Cloud, "cloud", cloudAWS, "Cloud provider: aws(default)|gcp)")
+	flags.StringVar(&opt.CredsFile, "creds-file", "", "Cloud credentials file (defaults vary depending on cloud)")
+	return cmd
+}
+
+// Complete finishes parsing arguments for the command
+func (o *Options) Complete(cmd *cobra.Command, args []string) error {
+	o.homeDir = "."
+
+	if u, err := user.Current(); err == nil {
+		o.homeDir = u.HomeDir
+	}
+	return nil
+}
+
+// Validate ensures that option values make sense
+func (o *Options) Validate(cmd *cobra.Command) error {
+	return nil
+}
+
+// Run executes the command
+func (o *Options) Run(dynClient client.Client, args []string) error {
+	if err := apis.AddToScheme(scheme.Scheme); err != nil {
+		return err
+	}
+	rh, err := o.getResourceHelper()
+	if err != nil {
+		return err
+	}
+
+	// Update the current HiveConfig, which should always exist as the operator will
+	// create a default one once run.
+	hc := &hivev1.HiveConfig{}
+	err = dynClient.Get(context.Background(), types.NamespacedName{Name: "hive"}, hc)
+	if err != nil {
+		log.WithError(err).Fatal("error looking up HiveConfig 'hive'")
+	}
+
+	hc.Spec.ManagedDomains = args
+
+	if o.Cloud == cloudAWS {
+		// Apply a secret for credentials to manage the root domain:
+		credsSecret, err := o.generateAWSCredentialsSecret()
+		if err != nil {
+			log.WithError(err).Fatal("error generating manageDNS credentials secret")
+		}
+		log.Infof("created Route53 credentials secret: %s", credsSecret.Name)
+		credsSecret.Namespace = hiveNamespace
+		rh.ApplyRuntimeObject(credsSecret, scheme.Scheme)
+
+		hc.Spec.ExternalDNS = &hivev1.ExternalDNSConfig{
+			AWS: &hivev1.ExternalDNSAWSConfig{
+				Credentials: corev1.LocalObjectReference{Name: manageDNSCredentialsSecret},
+			},
+		}
+	}
+
+	err = dynClient.Update(context.Background(), hc)
+	if err != nil {
+		log.WithError(err).Fatal("error updating HiveConfig")
+	}
+	log.Info("updated HiveConfig")
+
+	// Adding manageDNS to HiveConfig triggers a new rollout of the hiveadmission pods.
+	// To know when it's safe to proceed, we will list all hiveadmission pods and ensure
+	// they all have the expected env var set.
+	log.Info("waiting for new hiveadmission pods to deploy")
+	selector := map[string]string{"hiveadmission": "true"}
+	for i := 0; i < 24; i++ {
```

Comment:
I would prefer to see this done with a `UntilWithSync` from k8s.io/client-go/tools/watch. It does not matter too much, though.

Reply:
Looks useful will try to remember that in future.